admin-ui/plugins/auth-server/components/AuthN/DefaultAcr.js

```diff
@@ -0,0 +1,134 @@
+
+  const { t } = useTranslation()
+
+  const [modal, setModal] = useState(false)
+  const [put, setPut] = useState(null)
+  const userAction = {}
+  const theme = useContext(ThemeContext)
+  const selectedTheme = theme.state.theme
+
+  SetTitle('ACR Management')
+
+  useEffect(() => {
+    const initializeAcr = async () => {
+      try {
+        await authorize([ACR_WRITE, ACR_READ])
+      } catch (error) {
+        console.error('Error authorizing ACR permissions:', error)
+      }
+    }
+
+    initializeAcr()
+    dispatch(getScripts({ action: userAction }))
+    dispatch(getAgama())
+    dispatch(getAcrsConfig())
+  }, [authorize, dispatch])
+
+  const authScripts = useMemo(() => {
+    const filteredScripts = Array.isArray(scripts)
+      ? scripts
+          .filter((item) => item && item.scriptType === 'person_authentication' && item.enabled)
+          .map((item) => item.name)
+          .filter(Boolean)
+      : []
+
+    // const agamaFlows = Array.isArray(agamaList)
+    //   ? agamaList.map((flow) => flow?.details?.projectMetadata?.projectName).filter(Boolean)
+    //   : []
+    const agamaFlows = []
+
+    const result = [...filteredScripts, SIMPLE_PASSWORD_AUTH, ...agamaFlows]
+
+    console.log('Auth scripts with Agama flows:', result)
+    return result
+  }, [scripts, agamaList])
+
+  const toggle = () => {
+    setModal(!modal)
+  }
+
+  const putHandler = (putData) => {
+    setPut(putData)
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    setModal(true)
+  }
+
+  const handleSaveClick = () => {
+    setModal(true)
+  }
+
+  const submitForm = (userMessage) => {
+    toggle()
+
+    if (put?.value) {
+      const opts = {}
+      opts['authenticationMethod'] = { defaultAcr: put?.value }
+
+      buildPayload(userAction, userMessage, opts)
+      dispatch(editAcrs({ data: opts, action: userAction }))
+    } else {
+      console.error('No ACR value to save')
+    }
+  }
+
+  return (
+    <GluuLoader blocking={acrLoading || scriptLoading || agamaLoading}>
+      <Form onSubmit={handleSubmit}>
+        <GluuCommitDialog handler={toggle} modal={modal} onAccept={submitForm} />
+        <div style={{ padding: '3vh' }}>
+          <GluuViewWrapper canShow={hasCedarPermission(ACR_READ)}>
+            <DefaultAcrInput
+              id="defaultAcr"
+              name="defaultAcr"
+              lsize={6}
+              rsize={6}
+              type="select"
+              label={t('fields.default_acr')}
+              handler={putHandler}
+              value={acrs?.defaultAcr}
+              options={authScripts}
+              path={'/ACR'}
+              showSaveButtons={false}
+            />
+          </GluuViewWrapper>
+          {hasCedarPermission(ACR_WRITE) && (
+            <Button color={`primary-${selectedTheme}`} onClick={handleSaveClick}>
+              <i className="fa fa-check-circle me-2"></i>
+              {t('actions.save')}
+            </Button>
+          )}
```

Comment:
Disable Save when unchanged or empty.

Prevents no-op commits and “No ACR value to save” errors.

```diff
-            <Button color={`primary-${selectedTheme}`} onClick={handleSaveClick}>
+            <Button
+              color={`primary-${selectedTheme}`}
+              onClick={handleSaveClick}
+              disabled={!put?.value || put?.value === acrs?.defaultAcr}
+            >
```